pkg/cloudevents/client/defaulters_test.go

```diff
@@ -6,60 +6,44 @@ import (
 	"time"
 
 	"github.com/cloudevents/sdk-go/pkg/cloudevents"
-	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
 	"github.com/google/go-cmp/cmp"
 )
 
-func TestDefaultIDToUUIDIfNotSet(t *testing.T) {
-	testCases := map[string]struct {
-		event cloudevents.Event
-	}{
-		"nil context": {
-			event: cloudevents.Event{},
-		},
-		"v0.1 empty": {
-			event: cloudevents.Event{
-				Context: &cloudevents.EventContextV01{},
-			},
-		},
-		"v0.2 empty": {
-			event: cloudevents.Event{
-				Context: &cloudevents.EventContextV02{},
-			},
-		},
-		"v0.3 empty": {
-			event: cloudevents.Event{
-				Context: &cloudevents.EventContextV03{},
-			},
-		},
-		"v0.1 no change": {
-			event: cloudevents.Event{
-				Context: cloudevents.EventContextV01{EventID: "abc"}.AsV01(),
-			},
-		},
-		"v0.2 no change": {
-			event: cloudevents.Event{
-				Context: cloudevents.EventContextV02{ID: "abc"}.AsV02(),
-			},
-		},
-		"v0.3 no change": {
-			event: cloudevents.Event{
-				Context: cloudevents.EventContextV03{ID: "abc"}.AsV03(),
-			},
-		},
-	}
-	for n, tc := range testCases {
-		t.Run(n, func(t *testing.T) {
-
-			got := DefaultIDToUUIDIfNotSet(context.TODO(), tc.event)
-
-			if got.Context != nil && got.Context.AsV02().ID == "" {
+func TestDefaultIDToUUIDIfNotSet_empty(t *testing.T) {
+	for _, tc := range []string{"0.1", "0.2", "0.3", "1.0"} {
```

Comment:
would it be useful to hoist these into a var:
var versions = []string{"0.1", "0.2", "0.3", "1.0"}

Then in these loops use:
for _, version := range versions {